README.md

```diff
@@ -2,37 +2,109 @@
 
 ## Installation
 
-TODO
+These installation instructions assume deployment to a `Linux` or `MacOS` platform.
 
-* `virtualenv` instructions
-* `conda`/`miniconda`/`micromamba?` instructions
+`Windows` is currently **unsupported**.
 
-## User documentation
+### Downloading `um2netcdf-standalone`
 
-TODO
+```commandline
+cd <your projects dir>
+git clone https://github.com/ACCESS-NRI/um2nc-standalone.git
+cd um2nc-standalone
+```
+
+### Installing Dependencies
+
+#### MacOS
+
+```Bash
+$ brew install udunits
+```
+
+if this fails with some compiler/header errors, try:
+
+```Bash
+# set up non standard cf-unit paths
+$ export UDUNITS2_XML_PATH=/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2.xml
+$ export UDUNITS2_LIBDIR=/opt/homebrew/lib
+$ export UDUNITS2_INCDIR=/opt/homebrew/Cellar/udunits/2.2.28/include
+$ brew install udunits
+```
+
+#### Linux
+
+```Bash
+$ apt-get install udunits
+```
+
+**TODO:** add `yum` commands?
+
+## Creating a Python Runtime Environment
+
+The following instructions outline two options for creating a Python runtime environment fot `um2netcdf-standalone`. The purpose of a runtime environment is to manage specific library dependencies, separately to the system python.
+
+`virtualenv` is a low level python environment manager, with a companion `virtualenvwrapper` project to provide greater usability. These packages are available in `apt` for Linux and `homebrew` on Mac. The following instructions assume use of `virtualenvwrapper` ([virtualenvwrapper docs](https://virtualenvwrapper.readthedocs.io/en/latest/)).
+
+`conda` and related tools such as `miniconda` & `micromamba` are higher level environment managers than `virtualenv.
+
+If you are unsure which environment manager to use and/or new to Python, `conda` is the recommended approach. 
+
+### Creating a `virtualenv` environment
+
+```Bash
+# assuming virtualenvwrapper has been installed & configured on your system...
+# e.g. "brew install virtualenvwrapper" or "apt-get install virtualenvwrapper"
+$ mkvirtualenv -r requirements.txt ums
+
+# should download & install multiple packages
+# the command prompt should change with a (ums) prefix 
+```
+
+### Creating a `conda` environment
+
+```Bash
+# works for Linux and MacOS
+$ conda create -n ums
+$ activate ums
+$ conda install pip
+$ 
+$ cd <your-um2netcdf-project-dir>
+$ pip install -r requirements.txt
+```
 
 ### Running the tests
 
-This project uses `pytest`. To run the tests:
+The `um2nc-standalone` project uses `pytest`. To run the tests:
 
 ```Bash
 $ cd <your um2nc-standalone dir>
-$ pytest
+$ pytest  # should pass within seconds, possibly with warnings
 ```
 
-A minimal code coverage setup has been included, to run & generate an HTML coverage report:
+A minimal code coverage setup has been included, to run the tests & generate an HTML coverage report:
 
-```
+```Bash
 $ cd <your um2nc-standalone dir>
 $ pytest --cov-report=html --cov=umpost
```

Comment:
I've made it to this point and hit a few snags:

- The tests don't correctly load the source code for testing. I've made some changes to the import statements that appear to fix the issue.
- There's reference in both the tests and the source to a module `um2netcdf4` - however, the `umpost` "module" contains only a straight `um2netcdf` (missing the trailing `4`). I'm assuming it's meant to be referencing the one in this code base?

I'm going to make a wild assumption that what's happened is this code is being tested on a system where there is already some version of things called `umpost` and `um2netcdf4` installed, and `pytest` is actually testing those rather than the code in the repo...

Reply:
Yes, we had an odd temporary arrangement going:
* I developed `umpost/um2netcdf`
* Spencer developed the conversion driver script on `gadi`, using an environment with `um2netcdf4` & testing against that

#80 covers an import switchover to `umpost/um2netcdf`, which got merged 5 minutes ago. Do you want to grab the latest `develop` branch & see if you can get a working env & run the tests?

Reply:
I'm looking at the latest `develop` branch, but it's still angry about trying to find `umpost`:
```
ImportError while importing test module '/Users/marc/Documents/ACCESS-NRI/um2nc-standalone/test/test_um2netcdf.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/ums/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_um2netcdf.py:5: in <module>
    import umpost.um2netcdf as um2nc
E   ModuleNotFoundError: No module named 'umpost'
```

Reply:
Looks like the impasse may come from `pytest` vs `python -m pytest` - the latter works fine for me

https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html#invoking-pytest-versus-python-m-pytest

Reply:
I wonder if that's due to `python -m` adding `cwd` to `PYTHONPATH` behind the scenes, whereas `pytest` alone doesn't do this. If I start a fresh terminal & activate the `um2nc` venv, I have to export/set the `PYTHONPATH` manually in that term session.

Reply:
That's exactly the issue. There is a config setting for `pyproject.toml` that sets the `pythonpath` for `pytest`, but I can't seem to make it work properly for the current directory structure...